fud2/src/main.rs

```diff
@@ -46,15 +51,34 @@ fn build_driver() -> Driver {
         Ok(())
     });
     bld.rule(&[dahlia_setup], dahlia, calyx, "dahlia-to-calyx");
+    (dahlia, dahlia_setup)
+}
 
-    // MrXL.
+fn setup_mrxl(
+    bld: &mut DriverBuilder,
+    calyx: StateRef,
+) -> (StateRef, SetupRef) {
     let mrxl = bld.state("mrxl", &["mrxl"]);
     let mrxl_setup = bld.setup("MrXL compiler", |e| {
         e.var("mrxl-exe", "mrxl")?;
         e.rule("mrxl-to-calyx", "$mrxl-exe $in > $out")?;
         Ok(())
     });
     bld.rule(&[mrxl_setup], mrxl, calyx, "mrxl-to-calyx");
+    (mrxl, mrxl_setup)
+}
+
+fn build_driver() -> Driver {
+    let mut bld = DriverBuilder::new("fud2");
+
+    // The verilog state
+    let verilog = bld.state("verilog", &["sv", "v"]);
+    // Calyx.
+    let (calyx, calyx_setup) = setup_calyx(&mut bld, verilog);
```

Comment:
This looks great for now! FWIW, one way I was thinking of trying to make this stuff a bit more modular in the future was to add a, like, `DriverBuilder::get_state` function to look up a state by its name. That way, these individual "build a chunk of functionality" pieces wouldn't need the `StateRef`s threaded through, which could be a teensy bit more convenient. No need to do this here though!!

Reply:
Yup, that sounds like a good idea!